float32 vel_x
float32 vel_y
float32 position_x
float32 position_y
